File: benchmarks/js/fib.js
// Equivalent JavaScript implementation for benchmarking
// Run with: bun fib.js

function fib(n) {
  if (n <= 1) {
    return n;
  }
  return fib(n - 1) + fib(n - 2);
}

function main() {
  const n = 40;
  const result = fib(n);
  console.log(result);
}

main();

File: benchmarks/rust/fib.rs
// Equivalent Rust implementation for benchmarking
// Compile with: rustc -O -C target-cpu=native fib.rs

fn fib(n: i64) -> i64 {
    if n <= 1 {
        return n;
    }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    let n = 40;
    let result = fib(n);
    println!("{}", result);
}
